app/reflexes/admin/orders_reflex.rb

```diff
@@ -96,5 +99,16 @@ def success(i18n_key, count)
     def editable_orders
       Permissions::Order.new(current_user).editable_orders
     end
+
+    def all_distributors_can_invoice?(orders)
+      distributors = orders.map(&:distributor).uniq.reject(&:can_invoice?)
```

Comment:
This seems to be quite expensive for a bulk operation. It may load a distributor from the database for each order. I think that we can do this in a single SQL query. The easiest would probably be to add a `can_invoice` scope to the Enterprise model.